### PR TITLE
Post review fixes

### PR DESCRIPTION
**Pull request by ElDeveloper**
Most of these changes came up after I met with MacKenzie to go over the notebook. In summary:

- There was a bug with merging Bioinformatics/Contact sections.
- Merging sample sheets should ignore date differences.
- I updated the wording in the amplicon notebook to refer to preparations files as mapping files.
- Added tests for the seqpro CLI. Looks like the updates to the API didn't make it there so I added a new test module and updated the API.

### CODE REVIEW

**Review by charles-cowart — APPROVED** [review]

**Comment by charles-cowart** [comment]
@ElDeveloper everything looks good to me. I'll hold off on merging until @antgonza  looks at it.

File: metapool/sample_sheet.py [hunk 1]
```diff
@@ -280,24 +280,34 @@ def write_blank_lines(writer, n=blank_lines, width=csv_width):
     def merge(self, sheets):
         """Merge the Data section of multiple sample sheets
 
+        For the Date field in the Header section, we only keep the date of the
```
Comment (antgonza) [inline metapool/sample_sheet.py:283]: Line 281 says Data, here it says Date, are both correct?

File: metapool/sample_sheet.py [hunk 2]
```diff
@@ -280,24 +280,34 @@ def write_blank_lines(writer, n=blank_lines, width=csv_width):
...
         ValueError
-            If the header section is different between merged sheets.
+            If the Header, Settings or Reads section is different between
+            merged sheets.
         """
         for number, sheet in enumerate(sheets):
             for section in ['Header', 'Settings', 'Reads']:
-                if getattr(self, section) != getattr(sheet, section):
+
+                # For the Header section we'll ignore the Date field since that
```
Comment (antgonza) [inline metapool/sample_sheet.py:300]: I guess it's Date ...

File: metapool/sample_sheet.py [hunk 3]
```diff
@@ -280,24 +280,34 @@ def write_blank_lines(writer, n=blank_lines, width=csv_width):
...
+                # is likely to be different but shouldn't be a condition to
+                # merge two sheets.
+                this, that = getattr(self, section), getattr(sheet, section)
```
Comment (antgonza) [inline metapool/sample_sheet.py:303]: If this is only needed if sections is == 'Header', should it be under the if?

**Review by antgonza — COMMENTED** [review]
... a few comments ...

File: metapool/sample_sheet.py (hunk 1 above)
Comment (ElDeveloper) [reply metapool/sample_sheet.py:283]: You are right, it is Date. Date is a field in the Header section and Data is a different section. Hierarchy is: Section > Field

File: metapool/sample_sheet.py (hunk 3 above)
Comment (ElDeveloper) [reply metapool/sample_sheet.py]: My bad, that comment section was in the wrong location. It is still handy to have `this` and `that`, but the Date check is specific for the Header section. I've reordered this, should be clearer now.

**Comment by ElDeveloper** [comment]
This should be ready for review/merge.

**Comment by ElDeveloper** [comment]
Thanks @antgonza and @charles-cowart

**Pull request merged**